scipy/sparse/data.py

```diff
@@ -270,23 +289,29 @@ def _arg_min_or_max(self, axis, out, op, compare):
                 am = op(mat.data)
                 m = mat.data[am]
 
-                if compare(m, zero):
-                    return mat.row[am] * mat.shape[1] + mat.col[am]
-                else:
-                    size = np.prod(mat.shape)
-                    if size == mat.nnz:
+                if nonzero:
+                    if m == 0:
+                        return []
+                    else:
                         return am
+                else:
```

Comment:
Since the if-block always returns, I'd drop the else here and preserve the original indentation.

Reply:
I'm not sure what you're thinking here. The if block was introduced to bypass the compare operation, otherwise the zero would be considered.

Reply:
Sorry I wasn't clear. What I meant was:

```python
if nonzero:
    if m == 0:
        return []
    return am

if compare(m, zero):
    return mat.row...

...etc...
```

Reply:
Thanks. I understand now that the elses are redundant. I made those changes.